src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,483 @@
+    # Use the hour field to create a datetime field, switch to 0-23hr notation.
+    fixed_df = fixed_df.reset_index()
+    fixed_df["local_time"] = fixed_df.plan_date + \
+        pd.to_timedelta(fixed_df.hour - 1, unit="h")
+
+    # Find and replace the 0.0 values at 2-3am on the spring_forward days.
+    spring_forward_dates = list(pd.to_datetime(spring_forward).date)  # noqa: F841
+    demand_gaps = (
+        fixed_df
+        .query("plan_date in @spring_forward_dates")
+        .query("hour in [2,3]")
+        .query("demand_mwh==0.0")
+    )
+    fixed_df.loc[demand_gaps.index, "demand_mwh"] = np.nan
+    # Now that we have a real datetime column, ditch "hours" and "plan_date"
+    fixed_df = fixed_df.loc[:, ["respondent_id", "local_time",
+                                "timezone", "demand_mwh"]]
+
+    return fixed_df
+
+
+def _fix_tz_by_date(df, time_changes, respondent_id,
+                    start_time, end_time, tz_col, std_val, dst_val):
+    """Assign timezone abbreviations based on date."""
+    df = df.copy()
+    years = df.loc[(df.respondent_id == respondent_id) &
+                   (df.local_time >= start_time) &
+                   (df.local_time <= end_time)].local_time.dt.year.unique()
+    for yr in years:
+        spring = time_changes.loc[yr, "spring_forward"]
+        fall = time_changes.loc[yr, "fall_back"]
+        std_times = (
+            (df.respondent_id == respondent_id) &
+            (df.local_time.dt.year == yr) &
+            ((df.local_time < spring) | (df.local_time >= fall))
+        )
+        dst_times = (
+            (df.respondent_id == respondent_id) &
+            (df.local_time.dt.year == yr) &
+            ((df.local_time >= spring) & (df.local_time < fall))
+        )
+        df.loc[std_times, tz_col] = std_val
+        df.loc[dst_times, tz_col] = dst_val
+    return df
+
+
+def _apply_tz_fixes_by_date(df, fixes, time_changes):
+    """A wrapper assin many timezone fixes at once."""
+    for fix in fixes:
+        df = _fix_tz_by_date(df, time_changes=time_changes, **fix)
+    return df
+
+
+def respondents_ferc714(pudl_settings):
+    """Table of FERC 714 respondents IDs, names, and EIA utility IDs."""
+    zippath = zipfile.Path(
+        pathlib.Path(pudl_settings["data_dir"], "ferc714/data/ferc714.zip"),
+        "Respondent IDs.csv"
+    )
+    with zippath.open() as zp:
+        df = pd.read_csv(zp)
+    df = (
+        df.astype({
+            "eia_code": pd.Int64Dtype(),
+            "respondent_id": pd.Int64Dtype(),
+            "respondent_name": pd.StringDtype(),
+        })
+        .assign(
+            respondent_name=lambda x: x.respondent_name.str.strip(),
+            eia_code=lambda x: x.eia_code.replace(to_replace=0, value=pd.NA)
+        )
+        # These excludes fake Test IDs -- not real planning areas
+        .query("respondent_id not in @bad_respondents")
+    )
+    return df
+
+
+def planning_area_hourly_demand_ferc714(pudl_settings):
```

Comment:
it would be nice for me to have some in line comments of *why* you are doing some of these things in here. but in general, this looks really clean.

Reply:
Yes, sorry about the lack of docstrings and comments. I just wanted to get something pushed. I will flesh it out and make sure there's more explanation before asking for reviews next time.